web/packages/teleport/src/Navigation/SideNavigation/Section.tsx

```diff
@@ -18,37 +18,136 @@
 
 import React from 'react';
 import { NavLink } from 'react-router-dom';
-import styled, { css } from 'styled-components';
+import styled, { css, useTheme } from 'styled-components';
 
-import { Box } from 'design';
+import { Box, ButtonIcon, Flex, P2, Text } from 'design';
 import { Theme } from 'design/theme';
+import { ArrowLineLeft } from 'design/Icon';
+import { HoverTooltip, ToolTipInfo } from 'shared/components/ToolTip';
 
-import { NavigationSection } from './Navigation';
+import cfg from 'teleport/config';
+
+import { NavigationSection, NavigationSubsection } from './Navigation';
 import { zIndexMap } from './zIndexMap';
 import { CategoryIcon } from './CategoryIcon';
 
-export function Section({
+type SharedSectionProps = {
+  section: NavigationSection;
+  $active: boolean;
+  isExpanded: boolean;
+  onExpandSection: () => void;
+};
+
+/**
+ * CustomChildrenSectionProps is a type for a NavigationSection with custom children (e.g. search section).
+ */
+type CustomChildrenSectionProps = SharedSectionProps & {
+  children: React.ReactNode;
+};
```

Comment:
I'd remove this and have the props of the `CustomChildrenSection` component be `PropsWithChildren<SharedSectionProps>`